src/OpenSpaceToolkit/Astrodynamics/Trajectory.cpp

```diff
@@ -109,16 +109,56 @@ State                           Trajectory::getStateAt                      (
 Array<State>                    Trajectory::getStatesAt                     (   const   Array<Instant>&             anInstantArray                              ) const
 {
 
-    Array<State> stateArray = Array<State>::Empty() ;
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Trajectory") ;
+    }
+
+    // Check if array is empty, and if not then sort it before it is looped through
+    if (anInstantArray.isEmpty())
+    {
+        throw ostk::core::error::runtime::Undefined("Empty Instant Array supplied") ;
+    }
+
+    Array<std::pair<Instant, size_t>> instantArrayPair  = Array<std::pair<Instant, size_t>>::Empty() ;
+    instantArrayPair.reserve(anInstantArray.getSize()) ;
+
+    for (size_t i = 0 ; i < anInstantArray.getSize() ; i++)
+    {
+        instantArrayPair.push_back(std::make_pair(anInstantArray[i], i));
+    }
+
+    // Sort instant array pair chronologically
+    std::sort(instantArrayPair.begin(), instantArrayPair.end(), [] (const auto& lhs, const auto& rhs) { return lhs.first < rhs.first ; }) ;
```

Comment:
Use clearer variables' name to align with style of the lib maybe (nit)

Reply:
fixed